README.md

```diff
@@ -259,30 +237,41 @@ ListenerRegistration listenerRegistration = account.blockchainEvents()
                 }
             });
 ```
-For unregister the listener use `listenerRegistration.remove()` method.
+
+### Listening to balance changes
+
+Account balance changes, can be observed by adding balance listener using `BlockchainEvents`:
+
+```java
+ListenerRegistration listenerRegistration = account.addBalanceListener(new EventListener<Balance>() {
+            @Override
+            public void onEvent(Balance balance) {
+                Log.d("example", "balance event, new balance is = " + balance.value().toPlainString());
+            }
+        });
+```
 
 ### Listening to account creation
+
 Account creation on the blockchain network, can be observed, by adding create account listener using `BlockchainEvents`:
 
 ```java
-ListenerRegistration listenerRegistration = account.blockchainEvents()
-            .addAccountCreationListener(new EventListener<Void>() {
-                @Override
-                public void onEvent(Void result) {
-                    Log.d("example", "Account has created.);                     
-                }
-            });
+ListenerRegistration listenerRegistration = account.addAccountCreationListener(new EventListener<Void>() {
+            @Override
+            public void onEvent(Void result) {
+                Log.d("example", "Account has created.");
+            }
+        });
 ```
-For unregister the listener use `listenerRegistration.remove()` method.
 
-### Sync vs Async
+For unregister any listener use `listenerRegistration.remove()` method.
 
-Asynchronous requests are supported by our `Request` object. The `request.run()` method will perform the request on a serial 
-background thread and notify success/failure using `ResultCallback` on the android main thread. 
-In addition, `cancel(boolean)` method can be used to safely cancel requests and detach callbacks.
+## Sync vs Async
 
+Asynchronous requests are supported by our `Request` object. The `request.run()` method will perform the requests sequentially on a single background thread and notify success/failure using `ResultCallback` on the android main thread.
+In addition, `cancel(boolean)` method can be used to safely cancel requests and detach callbacks.
```

Comment:
why cancel gets a boolean paramter?

Reply:
`true if the request should be interrupted; otherwise, in-progress requests are
 allowed to complete`
maybe it's redundant and should interrupted always.